src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -59,18 +62,121 @@ def standardize_patient_name(
+
+    Args:
+        resource (dict): A FHIR resource respresented as a dictionary.
+        code_type (str): A string equal to 'alpha_2', 'alpha_3', or 'numeric' to specify
+                         which type of standard country identifier to generate.
+    Returns:
+        countries (list): A list containing country codes as specified by code_type for
+        each country found in resource.
+
+    We currently only handle extracting countries from patient resources, however as
+    need arises is the future funcionality can be expanded.
+    """
+    countries = []
+    resource_type = resource.get("resource").get("resourceType")
+    if resource_type == "Patient":
+        for address in resource.get("resource").get("address"):
+            country = address.get("country")
+            countries.append(standardize_country(country, code_type))
+    return countries
+
 
-    >>> standardize_phone("(555) 555-1212")
-    '5555551212'
+def standardize_phone(raw: str, countries: List = [None, "US"]) -> str:
+    """
+    Given a phone number and optionally an associated FHIR resource and country
+    extraction function if able to parse the phone number return it in the E.164
+    standardard international format. If the phone number is not parseable return none.
+
+    Phone number parsing process:
+    A maximum of three attemtps are made to parse any phone number.
+    First, we try to parse the phone number without any additional country information.
+    If this succeeds it must have been provided in a standard inernational format which
+    is the ideal case. If the first attempt fails and a FHIR resource associated with
+    the phone number and a country extraction function are provided we make a second
+    attempt to parse the phone number using any countries extracted from the resource.
+    Finally, in the case where the second attempt fails for a resource and country
+    extraction function are not provided we make a final attempt to parse the number
+    assuming it is American.
+
+    Args:
+        raw (str): Raw phone number to be standardized.
+
+        resource (dict): FHIR resource that the raw phone originated from that possibly
+                         contains information indicating which country the phone number
+                         is from.
+
+        countries (list): A list containing 2 letter ISO country codes for each country
+                          extracted from resource of the phone number to be standardized
+                          that might indicate it the phone numbers country of origin.
+    Returns:
+        standardized (str): The standardized phone number in E.164 format when the raw
+                            phone number was succesfully parsed and an emptry string
+                            otherwise.
     """
 
-    raw = [x for x in raw if x.isnumeric()]
-    raw = "".join(raw)
-    if len(raw) != 10:
-        raw = None
-    return raw
+    if countries != [None, "US"]:
```

Comment:
Q: do you think this would be more readable for the user if we did something like the following?

```python
def standardize_phone(raw: str, countries: List = []) -> str:
    <code>
    countries = [None] + countries + ["US"]
    <code>
```

The thought here is that it reduces the need for the `if` statement. I'm torn on whether or not it makes the code more readable though. I think either way is fine, just wanted to get your read on it.